python/plugins/processing/algs/qgis/ui/HeatmapWidgets.py

```diff
@@ -113,8 +116,9 @@ def pixelSizeChanged(self):
 
     def rowsChanged(self):
         rows = self.mRowsSpinBox.value()
-        if rows <= 0:
+        if rows <= 0 or self.raster_bounds.isNull():
             return
+
         cell_size = self.raster_bounds.height() / rows
         cols = max(round(self.raster_bounds.width() / cell_size) + 1, 1)
```

Comment:
Should a return be added here if cell_size = 0 ( would be if the raster_bounds are an horizontal line or a point )